core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -948,4 +976,16 @@ private static String msg(String left, String right) {
         }
         return "";
     }
+
+    /**
+     * This test ensures that queries that need to be rewritten have dedicated tests.
+     * These queries must override this method accordingly.
+     */
+    public void testMustRewrite() throws IOException {
```

Comment:
This one makes FunctionScoreQueryBuilderTests with this seed: 5FB367958E3049EF:EB5C14A8F6674F6F . that's because although you set allow unmapped fields to the context, function score needs to have the field mapped (see DecayFunctionBuilder#187).